experiments/trustchain/trustchain_module.py

```diff
@@ -102,9 +161,29 @@ def request_random_signature(self):
         verified_peers = list(self.overlay.network.verified_peers)
         self.request_signature_from_peer(choice(verified_peers), rand_up * 1024 * 1024, rand_down * 1024 * 1024)
 
+    def send_to_leader_peer(self, block_num):
+        leader_peer = self.overlay.network.verified_peers[0]
+        for _ in range(block_num):
+            rand_up = randint(1, 1000)
+            rand_down = randint(1, 1000)
+            self.request_signature_from_peer(leader_peer, rand_up, rand_down)
+
+    @experiment_callback
+    def start_spamming_leader_peer(self, block_num=100):
+        """
+            Send block_num of blocks per second to leader peer per second.
+            NUM_TX environment variable will be used instead of block_num if defined
+            :param block_num: number of blocks per sec to send to leader peer
+        """
+        if os.getenv('NUM_TX'):
+            block_num = int(os.getenv('NUM_TX'))
+
+        self.request_signatures_lc = LoopingCall(self.send_to_leader_peer, int(block_num))
+        self.request_signatures_lc.start(1)
+
     def request_signature_from_peer(self, peer, up, down):
         self._logger.info("%s: Requesting signature from peer: %s" % (self.my_id, peer))
-        transaction = {"up": up, "down": down}
+        transaction = {"up": up, "down": down, "from_peer": self.my_id, "to_peer": str(peer)}
```

Comment:
You are adding the string representation of a `Peer` object to the transaction. Please only include the (numeric) id of the peer.